media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java

```diff
@@ -28,17 +28,20 @@
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.util.HashMap;
+import java.util.Map;
 
 /**
 * A wrapper around android.media.MediaPlayer that allows the native code to use it.
 * See media/base/android/media_player_bridge.cc for the corresponding native code.
 */
 @JNINamespace("media")
 public class MediaPlayerBridge {
-
+    /**
+     * Give the host application a chance to take over the control.
+     */
     public static class ResourceLoadingFilter {
         public boolean shouldOverrideResourceLoading(
-                MediaPlayer mediaPlayer, Context context, Uri uri) {
+                MediaPlayer mediaPlayer, Context context, Uri uri, Map<String, String> headers) {
```

Comment:
Is `headers` a mandatory parameter? Why don't let the old constructor and the new one coexist?